opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala

```diff
@@ -98,6 +126,24 @@ object Tracing {
       r   <- createChildOf(old, spanName, spanKind).use(finalizeSpanUsingEffect(effect, _, toErrorStatus))
     } yield r
 
+  /**
+   * Sets the current span to be the child of the current span with name
+   * 'spanName', introducing a thread-local scope during the execution allowing
+   * non-zio context propagation. Ends the span (and scope) when the effect
+   * finishes.
```

Comment:
The intended behaviour will break if `effect` for some reason switches the ExecutionContext or contains an interruptible region. Maybe that should be noted in the Scaladoc.

Reply:
Good call. Interestingly enough, the java instrumentation package has advice for most implementations of execution context switching (akka, java util future, scala future, runnable, callable, etc). I've wondered if perhaps the better implementation of this would be some sort of zio-aware advice... Still trying to figure out how that all works.